File: php/73-set-matrix-zeroes.php
<?php
class Solution {

    /**
     * @param Integer[][] $matrix
     * @return NULL
     */
    function setZeroes(&$matrix) {
        $row = [];
        $col = [];
        $rowC = count($matrix);
        $colC = count($matrix[0]);
        
        foreach ($matrix as $k => $v) {
            foreach ($v as $kk => $vv) {
                if ($vv == 0) {
                    $row[] = $k;
                    $col[] = $kk;
                }
            }
        }
        foreach ($row as $r) {
            for ($i = 0; $i < $colC; $i++) {
                $matrix[$r][$i] = 0;
            }
        }
        foreach ($col as $c) {
            for ($i = 0; $i < $rowC; $i++) {
                $matrix[$i][$c] = 0;
            }
        }
        return;
    }
}

File: go/73-set-matrix-zeroes.go
package main

func setZeroes(matrix [][]int) {
	row := []int{}
	col := []int{}
	rlen := len(matrix)
	clen := len(matrix[0])
	for i := 0; i < rlen; i++ {
		for j := 0; j < clen; j++ {
			if matrix[i][j] == 0 {
				row = append(row, i)
				col = append(col, j)
			}
		}
	}
	for i := 0; i < len(row); i++ {
		for j := 0; j < clen; j++ {
			matrix[row[i]][j] = 0
		}
	}
	for i := 0; i < len(col); i++ {
		for j := 0; j < rlen; j++ {
			matrix[j][col[i]] = 0
		}
	}
}
